pgtoolkit/errors.py

```diff
@@ -1,12 +1,13 @@
 from __future__ import annotations
 
+from dataclasses import dataclass
 
+
+@dataclass
 class ParseError(Exception):
-    def __init__(self, lineno: int, line: str, message: str) -> None:
-        self.message = message
-        super().__init__(self.message)
-        self.lineno = lineno
-        self.line = line
+    lineno: int
+    line: str
+    message: str
 
     def __repr__(self) -> str:
```

Comment:
Since we use a `dataclass` do we still need the `__repr__`  method ? what about simply using the default one ?

> repr: If true (the default), a [__repr__()](https://docs.python.org/3/reference/datamodel.html#object.__repr__) method will be generated. The generated repr string will have the class name and the name and repr of each field, in the order they are defined in the class. Fields that are marked as being excluded from the repr are not included. For example: InventoryItem(name='widget', unit_price=3.0, quantity_on_hand=10).

Reply:
I agree. We're loosing the truncate of the message property but that's not a big deal. It would be a problem if it was removed from `__str__`.